diff --git a/dependencies/tool_metadata.yml b/dependencies/tool_metadata.yml
index d708143..140abca 100644
--- a/dependencies/tool_metadata.yml
+++ b/dependencies/tool_metadata.yml
@@ -58,7 +58,7 @@
   in_install: false
 - name: openroad_app
   repo: https://github.com/The-OpenROAD-Project/OpenROAD
-  commit: 4174c3ad802d2ac1d04d387d2c4b883903f6647e
+  commit: 5e557eeb13dcbc19a167fb7c00d06006db9c9433
   build: ''
   in_install: false
 - name: git
diff --git a/flow.tcl b/flow.tcl
index 8395418..2ec1686 100755
--- a/flow.tcl
+++ b/flow.tcl
@@ -156,7 +156,7 @@ proc run_klayout_step {args} {
 proc run_post_run_hooks {} {
     if { [file exists $::env(DESIGN_DIR)/hooks/post_run.py]} {
         puts_info "Running post run hook"
-        set result [exec $::env(OPENROAD_BIN) -python $::env(DESIGN_DIR)/hooks/post_run.py]
+        set result [exec $::env(OPENROAD_BIN) -exit -python $::env(DESIGN_DIR)/hooks/post_run.py]
         puts_info "$result"
     } else {
         puts_info "hooks/post_run.py not found, skipping"
diff --git a/scripts/tcl_commands/all.tcl b/scripts/tcl_commands/all.tcl
index 29efd3a..a02085b 100755
--- a/scripts/tcl_commands/all.tcl
+++ b/scripts/tcl_commands/all.tcl
@@ -120,7 +120,7 @@ proc prep_lefs {args} {
         if { [info exists ::env(METAL_LAYER_NAMES)] } {
             set ::env(TECH_METAL_LAYERS) $::env(METAL_LAYER_NAMES)
         } else {
-            try_catch $::env(OPENROAD_BIN) -python\
+            try_catch $::env(OPENROAD_BIN) -exit -python\
                 $::env(SCRIPTS_DIR)/odbpy/lefutil.py get_metal_layers\
                 -o $::env(TMP_DIR)/layers.list\
                 $arg_values(-tech_lef)
diff --git a/scripts/tcl_commands/eco.tcl b/scripts/tcl_commands/eco.tcl
index c258994..ea39271 100755
--- a/scripts/tcl_commands/eco.tcl
+++ b/scripts/tcl_commands/eco.tcl
@@ -65,6 +65,7 @@ proc eco_gen_buffer {args} {
     puts_info "\[ECO: $::env(ECO_ITER)\] Generating buffer insertion script..."
 
     try_catch $::env(OPENROAD_BIN) \
+	-exit \
         -python $::env(SCRIPTS_DIR)/odbpy/eco.py \
         insert_buffer \
         -o $::env(routing_tmpfiles)/eco_fix.tcl \
diff --git a/scripts/tcl_commands/floorplan.tcl b/scripts/tcl_commands/floorplan.tcl
index 1f7f5f3..9dc26a5 100755
--- a/scripts/tcl_commands/floorplan.tcl
+++ b/scripts/tcl_commands/floorplan.tcl
@@ -19,7 +19,7 @@ proc extract_core_dims {args} {
 
     set out_tmp $::env(TMP_DIR)/dimensions.txt
 
-    try_catch $::env(OPENROAD_BIN) -python $::env(SCRIPTS_DIR)/odbpy/defutil.py extract_core_dims\
+    try_catch $::env(OPENROAD_BIN) -exit -python $::env(SCRIPTS_DIR)/odbpy/defutil.py extract_core_dims\
         --output-data $out_tmp\
         --input-lef $::env(MERGED_LEF)\
         $::env(CURRENT_DEF)
@@ -77,7 +77,7 @@ proc init_floorplan {args} {
 
             set intermediate [index_file $::env(floorplan_tmpfiles)/minimized_pdn.txt]
 
-            try_catch $::env(OPENROAD_BIN) -python $::env(SCRIPTS_DIR)/odbpy/snap_to_grid.py\
+            try_catch $::env(OPENROAD_BIN) -exit -python $::env(SCRIPTS_DIR)/odbpy/snap_to_grid.py\
                 --output $intermediate\
                 --input-lef $::env(MERGED_LEF)\
                 [expr {$core_width/8.0}] [expr {$core_height/8.0}] [expr {$core_width/4.0}] [expr {$core_height/4.0}]
diff --git a/scripts/tcl_commands/klayout.tcl b/scripts/tcl_commands/klayout.tcl
index 3c889d3..21b8655 100755
--- a/scripts/tcl_commands/klayout.tcl
+++ b/scripts/tcl_commands/klayout.tcl
@@ -153,9 +153,9 @@ proc run_klayout_gds_xor {args} {
 					$arg_values(-layout1) $arg_values(-layout2) $::env(DESIGN_NAME) \
 					$arg_values(-output_gds) \
 					|& tee $::env(TERMINAL_OUTPUT) $log
-				try_catch $::env(OPENROAD_BIN) -python $::env(SCRIPTS_DIR)/parse_klayout_xor_log.py \
+				try_catch $::env(OPENROAD_BIN) -exit -python $::env(SCRIPTS_DIR)/parse_klayout_xor_log.py \
 					-l [index_file $::env(signoff_logs)/xor.log] \
-					-o [index_file $::env(signoff_reports)/xor.rpt]
+				        -o [index_file $::env(signoff_reports)/xor.rpt]
 				scrot_klayout -layout $arg_values(-output_gds) -log $::env(signoff_logs)/screenshot.klayout.xor.log
 			}
 
diff --git a/scripts/tcl_commands/lvs.tcl b/scripts/tcl_commands/lvs.tcl
index c6fc4ff..40c4cb2 100755
--- a/scripts/tcl_commands/lvs.tcl
+++ b/scripts/tcl_commands/lvs.tcl
@@ -93,7 +93,7 @@ proc write_powered_verilog {args} {
         set_if_unset arg_values(-powered_netlist) ""
     }
 
-    try_catch $::env(OPENROAD_BIN) -python $::env(SCRIPTS_DIR)/odbpy/power_utils.py write_powered_def\
+    try_catch $::env(OPENROAD_BIN) -exit -python $::env(SCRIPTS_DIR)/odbpy/power_utils.py write_powered_def\
         --output $arg_values(-output_def) \
         --input-lef $arg_values(-lef) \
         --power-port $arg_values(-power) \
diff --git a/scripts/tcl_commands/routing.tcl b/scripts/tcl_commands/routing.tcl
index c93aa51..2cbb37a 100755
--- a/scripts/tcl_commands/routing.tcl
+++ b/scripts/tcl_commands/routing.tcl
@@ -140,7 +140,7 @@ proc detailed_routing_tritonroute {args} {
     unset ::env(_tmp_drt_file_prefix)
     unset ::env(_tmp_drt_rpt_prefix)
 
-    try_catch $::env(OPENROAD_BIN) -python $::env(SCRIPTS_DIR)/drc_rosetta.py tr to_klayout \
+    try_catch $::env(OPENROAD_BIN) -exit -python $::env(SCRIPTS_DIR)/drc_rosetta.py tr to_klayout \
         -o $::env(routing_reports)/drt.klayout.xml \
         --design-name $::env(DESIGN_NAME) \
         $::env(routing_reports)/drt.drc
diff --git a/scripts/utils/utils.tcl b/scripts/utils/utils.tcl
index dc71b93..5f9e19e 100755
--- a/scripts/utils/utils.tcl
+++ b/scripts/utils/utils.tcl
@@ -424,7 +424,7 @@ proc manipulate_layout {args} {
     set_if_unset arg_values(-output) $arg_values(-input)
     set_if_unset arg_values(-output_def) /dev/null
 
-    try_catch $::env(OPENROAD_BIN) -python \
+    try_catch $::env(OPENROAD_BIN) -exit -python \
         {*}$args \
         --input-lef $::env(MERGED_LEF) \
         --output-def $arg_values(-output_def) \
diff --git a/tests/1007/interactive.tcl b/tests/1007/interactive.tcl
index 5f37101..2e7edf0 100644
--- a/tests/1007/interactive.tcl
+++ b/tests/1007/interactive.tcl
@@ -12,6 +12,6 @@ insert_buffer\
 
 exec cp $::env(CURRENT_ODB) $::env(DESIGN_DIR)/out.odb
 
-exec $::env(OPENROAD_BIN) -python $::env(DESIGN_DIR)/hooks/post_run.py
+exec $::env(OPENROAD_BIN) -exit -python $::env(DESIGN_DIR)/hooks/post_run.py
 
-puts_info "Done."
\ No newline at end of file
+puts_info "Done."
diff --git a/tests/892/hooks/post_run.py b/tests/892/hooks/post_run.py
index 9486dad..782464b 100644
--- a/tests/892/hooks/post_run.py
+++ b/tests/892/hooks/post_run.py
@@ -2,7 +2,7 @@
 # export MERGED_LEF=/openlane/designs/def_test/runs/RUN_2022.01.23_17.23.46/tmp/merged.lef
 # export HOOK_OUTPUT_PATH=/openlane/designs/def_test/runs/RUN_2022.01.23_17.23.46/results/final
 # export DESIGN_DIR=/openlane/designs/def_test
-# openroad -python designs/def_test/hooks/post_run.py
+# openroad -exit -python designs/def_test/hooks/post_run.py
 
 import odb
 
diff --git a/tests/__main__.py b/tests/__main__.py
index dd84770..5efa010 100755
--- a/tests/__main__.py
+++ b/tests/__main__.py
@@ -156,6 +156,7 @@ def run_test_case(test_case):
                 subprocess.run(
                     [
                         "openroad",
+                        "-exit",
                         "-python",
                         test_case_issue_regression_script,
                         os.path.join(test_case, "runs", "issue_regression_run"),
